app/src/main/java/com/example/tasky/feature_splash/domain/SplashViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.tasky.feature_splash.domain
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class SplashViewModel @Inject constructor() : ViewModel() {
+    // TODO check if user has an active session or needs to login
+    // todo if user has active session, navigate to the agenda screen, else to login screen
+
+    companion object {
+        private val TAG = "SplashViewModel"
+    }
+
+    var isUserLoggedIn = true
+
+    fun checkIfUserIsLoggedIn(): Boolean {
+        Log.d(TAG, "check if user is logged in")
+        return isUserLoggedIn
+    }
```

Comment:
If `isUserIsLoggedIn` is public, why have this extra function that returns its value? Why not just refer to `viewModel.isUserLoggedIn` from the UI?

Also, you might want to make that a Compose state or StateFlow, so you can listen to its changes in the Compose UI